### PR TITLE
Run CI tests in Release configuration

### PR DESCRIPTION
**Pull request by azarovalex**
Fixes: https://github.com/mapbox/mapbox-navigation-ios/issues/3742
This PR adds two new jobs that run tests in Release configuration with SPM and Carthage integrations respectively.
Some tests had to be disabled since they can only run in Debug configuration.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -677,19 +680,35 @@ workflows:
   main-workflow:
     jobs:
       - build-job:
-          name: "Xcode_12.5.1_iOS_14.5"
+          name: "Xcode_12.5.1_iOS_14.5_Debug"
```
Comment (MaximAlien) [inline .circleci/config.yml:683]: Should we take this as an opportunity to start using Xcode `v13.x` for some build jobs?

**Review by MaximAlien — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (azarovalex) [reply .circleci/config.yml:683]: Also thought about it, I'll try updating to Xcode 13, if we won't need a lot of changes I'll make it in this PR

File: .circleci/config.yml [hunk 2]
```diff
@@ -340,6 +340,9 @@ jobs:
       verify_localization:
         type: boolean
         default: true
+      scheme:
```
Comment (bamx23) [inline .circleci/config.yml:343]: Can we use `configuration` here? As `scheme` is used to be a different thing in terms of Xcode projects.

Comment (azarovalex) [reply .circleci/config.yml:343]: thx, it was a typo I promise 🙂

File: Tests/MapboxNavigationTests/SKUTests.swift [hunk 3]
```diff
@@ -6,6 +6,7 @@ import MapboxSpeech
 import MapboxNavigation
 import MapboxCommon_Private
 
+#if DEBUG
```
Comment (bamx23) [inline Tests/MapboxNavigationTests/SKUTests.swift:9]: Could you also add a comment on why we don't run some tests in the `Release` configuration?

Comment (azarovalex) [reply Tests/MapboxNavigationTests/SKUTests.swift:9]: Because the billing API used in these tests was already wrapped in #if DEBUG, for security reasons.

**Pull request merged**